File: week-3/peak-index-in-a-mountain-array/peak-index-in-a-mountain-array.py
class Solution:
    def peakIndexInMountainArray(self, A: List[int]) -> int:
        left = 0
        right = len(A) - 1
        while (left < right):
            mid = left + (right-left)//2
            if A[mid] < A[mid+1]:
                left = mid + 1
            else:
                right = mid
        
        return left

File: week-3/peak-index-in-a-mountain-array/peak-index-in-a-mountain-array.js
/**
 * @param {number[]} A
 * @return {number}
 */
var peakIndexInMountainArray = function(A) {
    let left = 0;
    let right = A.length-1;
    while (left < right) { 
        mid = left + Math.floor((right-left)/2);
        
        if (A[mid] < A[mid+1]) left = mid + 1;
        else right = mid
    }
    
    return left;
};
